internal/tracer/builder.go

```diff
@@ -116,7 +116,7 @@ func (b *builder) add(event Event) {
 		event.MessageIndex = b.reqCount
 		b.reqCount++
 	case *RequestBodyEnd:
-		if b.trace.Err != nil {
+		if b.trace.Err == nil {
```

Comment:
Oof, these error conditions were just straight-up wrong 🤦. The intent was to set the error if it's not already set. But the inverted condition meant it was _never_ being set.